diff --git a/Makefile.am b/Makefile.am
index e5f6203..7cea3c9 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,11 +1,9 @@
 SUBDIRS = \
 	libvips \
 	cplusplus \
-	tools \
 	po \
 	man \
 	doc \
-	test \
 	fuzz
 
 EXTRA_DIST = \
diff --git a/Makefile.in b/Makefile.in
index 2029423..45a3175 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -511,11 +511,9 @@ vips_introspection_sources = @vips_introspection_sources@
 SUBDIRS = \
 	libvips \
 	cplusplus \
-	tools \
 	po \
 	man \
 	doc \
-	test \
 	fuzz
 
 EXTRA_DIST = \
diff --git a/configure.ac b/configure.ac
index c983e81..1aa2e2b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -320,19 +320,7 @@ AC_HEADER_STDC
 AC_CHECK_HEADERS([errno.h math.h fcntl.h limits.h stdlib.h string.h sys/file.h sys/ioctl.h sys/param.h sys/time.h sys/mman.h sys/types.h sys/stat.h unistd.h io.h direct.h windows.h])
 
 # uncomment to change which libs we build
-# AC_DISABLE_SHARED
-# AC_DISABLE_STATIC
-AC_LIBTOOL_WIN32_DLL
-AC_CHECK_TOOL(DLLWRAP, dllwrap)
-AC_CHECK_TOOL(DLLTOOL, dlltool)
-AC_CHECK_TOOL(OBJDUMP, objdump)
-AC_CHECK_TOOL(RANLIB, ranlib)
-AC_CHECK_TOOL(STRIP, strip)
-AC_CHECK_TOOL(AR, ar)
-AC_CHECK_TOOL(AS, as)
-AC_CHECK_TOOL(LD, ld)
-AC_PROVIDE([AC_LIBTOOL_WIN32_DLL])
-AC_PROG_LIBTOOL
+AC_DISABLE_SHARED
 
 # Checks for typedefs, structures, and compiler characteristics.
 AC_C_RESTRICT
@@ -1348,16 +1336,6 @@ AC_CONFIG_FILES([
 	cplusplus/include/Makefile
 	cplusplus/include/vips/Makefile
 	cplusplus/Makefile
-	tools/Makefile
-	tools/batch_crop
-	tools/batch_image_convert
-	tools/batch_rubber_sheet
-	tools/light_correct
-	tools/shrink_width
-	test/Makefile
-	test/variables.sh
-	test/test-suite/Makefile
-	test/test-suite/helpers/Makefile
 	man/Makefile
 	doc/Makefile
 	doc/libvips-docs.xml
