diff --git a/build/vips-web.mk b/build/vips-web.mk
index db099bf..2101638 100644
--- a/build/vips-web.mk
+++ b/build/vips-web.mk
@@ -9,7 +9,8 @@ $(PKG)_GH_CONF  := libvips/libvips/releases,v
 $(PKG)_SUBDIR   := vips-$($(PKG)_VERSION)
 $(PKG)_FILE     := vips-$($(PKG)_VERSION).tar.gz
 $(PKG)_DEPS     := cc libwebp librsvg giflib glib pango libgsf \
-                   libjpeg-turbo tiff lcms libexif libpng libspng orc
+                   libjpeg-turbo tiff lcms libexif libpng libspng orc \
+                   imagemagick

 define $(PKG)_PRE_CONFIGURE
     # Copy some files to the packaging directory
@@ -29,8 +30,10 @@ define $(PKG)_PRE_CONFIGURE
      printf '  "glib": "$(glib_VERSION)",\n'; \
      printf '  "gsf": "$(libgsf_VERSION)",\n'; \
      printf '  "harfbuzz": "$(harfbuzz_VERSION)",\n'; \
+     printf '  "imagemagick": "$(imagemagick_VERSION)",\n'; \
      printf '  "jpeg": "$(libjpeg-turbo_VERSION)",\n'; \
      printf '  "lcms": "$(lcms_VERSION)",\n'; \
+     printf '  "openjpeg": "$(openjpeg_VERSION)",\n'; \
      printf '  "orc": "$(orc_VERSION)",\n'; \
      printf '  "pango": "$(pango_VERSION)",\n'; \
      printf '  "pixman": "$(pixman_VERSION)",\n'; \
@@ -60,7 +63,6 @@ define $(PKG)_BUILD
         $(MXE_DISABLE_DOC_OPTS) \
         --enable-debug=no \
         --without-fftw \
-        --without-magick \
         --without-heif \
         --without-openslide \
         --without-pdfium \
