diff --git a/build/overrides.mk b/build/overrides.mk
index d6eb112..80ba02f 100644
--- a/build/overrides.mk
+++ b/build/overrides.mk
@@ -478,8 +478,12 @@ define imagemagick_BUILD
 
     cd '$(BUILD_DIR)' && $(SOURCE_DIR)/configure \
         $(MXE_CONFIGURE_OPTS) \
+        --enable-static=yes \
+        --enable-shared=no \
+        --disable-shared \
         --without-fftw \
         --without-fontconfig \
+        --without-freetype \
         --without-gvc \
         --without-heic \
         --without-ltdl \
@@ -499,7 +503,7 @@ define imagemagick_BUILD
         --disable-largefile \
         --disable-opencl \
         --disable-openmp \
-        CFLAGS="$(CFLAGS) -DMAGICKCORE_EXCLUDE_DEPRECATED"
+        CFLAGS="-fPIC $(CFLAGS) -DMAGICKCORE_EXCLUDE_DEPRECATED"
     $(MAKE) -C '$(BUILD_DIR)' -j '$(JOBS)' $(MXE_DISABLE_CRUFT)
     $(MAKE) -C '$(BUILD_DIR)' -j 1 install $(MXE_DISABLE_CRUFT)
 endef
@@ -742,7 +746,8 @@ define glib_BUILD
     # build as shared library, since we need `libgobject-2.0-0.dll`
     # and `libglib-2.0-0.dll` for the language bindings.
     '$(TARGET)-meson' \
-        --default-library=shared \
+        --default-library=static \
+        -Dforce_posix_threads=true \
         --buildtype=release \
         --strip \
         --libdir='lib' \