diff --git a/build/package-vipsdev.sh b/build/package-vipsdev.sh
index 149c969..83fb5b1 100755
--- a/build/package-vipsdev.sh
+++ b/build/package-vipsdev.sh
@@ -85,12 +85,6 @@ whitelist=(api-ms-win-crt-{conio,convert,environment,filesystem,heap,locale,math
 whitelist+=(userenv.dll)
 
 # Copy libvips and dependencies with pe-util
-$mxe_prefix/$build_os/bin/peldd \
-  $mxe_prefix/$target.$deps/bin/$target_dll \
-  --clear-path \
-  --path $mxe_prefix/$target.$deps/bin \
-  ${whitelist[@]/#/--wlist } \
-  -a | xargs cp -t $repackage_dir/bin
 
 echo "Copying install area $mxe_prefix/$target.$deps/"
 
@@ -121,16 +115,6 @@ rm -rf $repackage_dir/share/locale
 # Remove those .gitkeep files
 rm $repackage_dir/{share,lib,include}/.gitkeep
 
-echo "Copying vips executables"
-
-# We still need to copy the vips executables
-cp $mxe_prefix/$target.$deps/bin/{vips,vipsedit,vipsheader,vipsthumbnail}.exe $repackage_dir/bin/
-
-echo "Strip unneeded symbols"
-
-# Remove all symbols that are not needed
-$mxe_prefix/bin/$target.$deps-strip --strip-unneeded $repackage_dir/bin/*.{exe,dll}
-
 echo "Copying packaging files"
 
 cp $mxe_prefix/$target.$deps/vips-packaging/{AUTHORS,ChangeLog,COPYING,README.md,versions.json} $repackage_dir