diff --git a/build/vips-web.mk b/build/vips-web.mk
index db099bf..eaefb23 100644
--- a/build/vips-web.mk
+++ b/build/vips-web.mk
@@ -9,7 +9,8 @@ $(PKG)_GH_CONF  := libvips/libvips/releases,v
 $(PKG)_SUBDIR   := vips-$($(PKG)_VERSION)
 $(PKG)_FILE     := vips-$($(PKG)_VERSION).tar.gz
 $(PKG)_DEPS     := cc libwebp librsvg giflib glib pango libgsf \
-                   libjpeg-turbo tiff lcms libexif libpng libspng orc
+                   libjpeg-turbo tiff lcms libexif libpng libspng orc \
+                   libheif imagemagick
 
 define $(PKG)_PRE_CONFIGURE
     # Copy some files to the packaging directory
@@ -18,6 +20,7 @@ define $(PKG)_PRE_CONFIGURE
 
     (printf '{\n'; \
      printf '  "cairo": "$(cairo_VERSION)",\n'; \
+     printf '  "de265": "$(libde265_VERSION)",\n'; \
      printf '  "exif": "$(libexif_VERSION)",\n'; \
      printf '  "expat": "$(expat_VERSION)",\n'; \
      printf '  "ffi": "$(libffi_VERSION)",\n'; \
@@ -29,8 +32,11 @@ define $(PKG)_PRE_CONFIGURE
      printf '  "glib": "$(glib_VERSION)",\n'; \
      printf '  "gsf": "$(libgsf_VERSION)",\n'; \
      printf '  "harfbuzz": "$(harfbuzz_VERSION)",\n'; \
+     printf '  "heif": "$(libheif_VERSION)",\n'; \
+     printf '  "imagemagick": "$(imagemagick_VERSION)",\n'; \
      printf '  "jpeg": "$(libjpeg-turbo_VERSION)",\n'; \
      printf '  "lcms": "$(lcms_VERSION)",\n'; \
+     printf '  "openjpeg": "$(openjpeg_VERSION)",\n'; \
      printf '  "orc": "$(orc_VERSION)",\n'; \
      printf '  "pango": "$(pango_VERSION)",\n'; \
      printf '  "pixman": "$(pixman_VERSION)",\n'; \
@@ -40,6 +46,7 @@ define $(PKG)_PRE_CONFIGURE
      printf '  "tiff": "$(tiff_VERSION)",\n'; \
      printf '  "vips": "$(vips-web_VERSION)",\n'; \
      printf '  "webp": "$(libwebp_VERSION)",\n'; \
+     printf '  "x265": "$(x265_VERSION)",\n'; \
      printf '  "xml": "$(libxml2_VERSION)",\n'; \
      printf '  "zlib": "$(zlib_VERSION)"\n'; \
      printf '}';) \
@@ -55,13 +62,14 @@ define $(PKG)_BUILD
         --host='$(TARGET)' \
         --build='$(BUILD)' \
         --prefix='$(PREFIX)/$(TARGET)' \
-        --disable-static \
-        --enable-shared \
+        --enable-static \
+        --disable-shared \
+        --disable-debug \
+        --disable-deprecated \
+        --disable-introspection \
         $(MXE_DISABLE_DOC_OPTS) \
         --enable-debug=no \
         --without-fftw \
-        --without-magick \
-        --without-heif \
         --without-openslide \
         --without-pdfium \
         --without-poppler \